content/drafts/onion-routing-prelims.md

```diff
@@ -0,0 +1,370 @@
+  which means that this is the number of sats that will be routed through 
+  Charlie. So the amount of fees to pay to Charlie can be calculated as follows:
+
+```
+ = Charlie’s Base Fee + Charlie’s proportional fee in parts per million/1000000 * (The amount routed through Charlie)
+ = 100 + 1000/1000000(4999999)  
+ = 100 + 4999999000/1000000
+ = 100 + 4999.999
+ = 5099.99
+ ~ 5100
+```
+
+So the total number of sats that should be sent to Charlie is: 
+```
+ = 4999999 + 5100 = 5005099 
+ = 5005099 msats
+```
+
+- 5005099 msats need to be routed through Bob, so the fee we would need to pay
+  to Bob is:
+ 
+```
+ = 200 + 2000/1000000(5005099)
+ = 200 + 2/1000(5005099)
+ = 10211
+```
+
+So the total number of sats to send to Bob is:
+```
+ = 5005099 + 10211
+ = 5015310
+```
+
+This means that Alice will pay the following amount in fees for the payment:
+```
+ = 5015310 - 4999999 
+ = 15311 msats
+```
+
+### Path 2
+
+Alice will do the same type of calculation for the path 2.
+
+- Charlie’s step will be the same: He must be paid a total of 5005099 msats.
+- Then we work out what we would need to pay Eve:
+ 
+```
+  = 300 + 3000/1000000(5005099)
+  = 300 + 3/1000(5005099)
+  = 15316 msats
+```
+
+So the total number of sats to send to Eve is:
+```
+  = 5005099 + 15316
+  = 5020415 msats
+```
+
+Ie, the total that Alice will pay in fees is: 
+```
+5020415-4999999 = 20416 msats
+```
+
+Aaaand the winner is: Path 1! Now we only need care about the following info:
+
+![](/onion_prelims/6-path1.png#center)
+
+# Hop Payloads
+
+Alice now has enough information to know what she wants to communicate to each 
+hop along the path.
+
+![](/onion_prelims/7-last-hop-info.png#center)
+
+Alice wants Dave to receive an `update_add_htlc` from Charlie on the `CD` 
+channel. The `payment_hash` should match the one specified in the invoice 
+received from Dave. The `amount_msat` will be 4999999 as specified in the 
+invoice. Alice then also checks the current block height (height 1001) and uses 
+that along with the `min_final_cltv_expiry_delta`(9)  in the invoice to 
+calculate that the outgoing CLTV value on the HTLC should be 1010.
+
+Using the fee calculation we did earlier for this path, Alice can also determine 
+what the `update_add_htlc`s should look like for the Bob-to-Charlie and 
+Alice-to-Bob steps. If the fact that the CLTVs for each HTLC increases from 
+right to left is confusing, check out [this](#a-note-on-cltv-deltas) quick CLTV 
+delta explainer. 
+
+![](/onion_prelims/8-other-hop-info.png#center)
+
+The only part of the puzzle that is missing here is: How does Alice tell 
+Charlie what to forward to Dave if she is only connected to Bob? This is where 
+the `onion_routing_packet` field in `update_add_htlc` comes in. Alice packages 
+up what she wants to tell each node and essentially then wraps it with the 
+payload for the previous node. The diagram below should make this more clear:
+
+![](/onion_prelims/9-onion-info.png#center)
```

Comment:
In the blue part of the diagram, I think it should say "Give to Charlie" instead of "Give to Dave" (same in the blog post cover image).